cycode/cli/commands/scan/code_scanner.py

```diff
@@ -439,12 +452,24 @@ def perform_scan(
 
 
 def perform_scan_async(
-    cycode_client: 'ScanClient', zipped_documents: 'InMemoryZip', scan_type: str, scan_parameters: dict
+    cycode_client: 'ScanClient',
+    zipped_documents: 'InMemoryZip',
+    scan_type: str,
+    scan_parameters: dict,
+    should_use_scan_service: bool = False,
 ) -> ZippedFileScanResult:
-    scan_async_result = cycode_client.zipped_file_scan_async(zipped_documents, scan_type, scan_parameters)
+    scan_async_result = cycode_client.zipped_file_scan_async(
+        zipped_documents, scan_type, scan_parameters, should_use_scan_service=should_use_scan_service
+    )
     logger.debug('scan request has been triggered successfully, scan id: %s', scan_async_result.scan_id)
 
-    return poll_scan_results(cycode_client, scan_async_result.scan_id, scan_type)
+    return poll_scan_results(
+        cycode_client,
+        scan_async_result.scan_id,
+        scan_type,
+        scan_parameters.get('report'),
+        should_use_scan_service=should_use_scan_service,
+    )
 
 
 def perform_commit_range_scan_async(
```

Comment:
async scans in terms of code naming here are scans that go to scan service. see the prev comment for more thoughts. imo this method should not be changed